homeassistant/components/freebox/number.py

```diff
@@ -0,0 +1,130 @@
+"""Support for Freebox Delta, Revolution and Mini 4K."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+
+from freebox_api.exceptions import InsufficientPermissionsError
+
+from homeassistant.components.number import (
+    NumberEntity,
+    NumberEntityDescription,
+    NumberMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .router import FreeboxRouter
+
+
+@dataclass
+class FreeboxNumberSetValueMixin:
+    """Mixin for required keys."""
+
+    async_set_value: Callable[[FreeboxRouter, int], Awaitable]
+
+
+@dataclass
+class FreeboxNumberEntityDescription(
+    NumberEntityDescription, FreeboxNumberSetValueMixin
+):
+    """Class describing Freebox number entities."""
+
+
+LCD_NUMBER_DESCRIPTIONS: tuple[FreeboxNumberEntityDescription, ...] = (
+    FreeboxNumberEntityDescription(
+        key="lcd_brightness",
+        name="LCD Brightness",
+        icon="mdi:monitor",
+        entity_category=EntityCategory.CONFIG,
+        native_min_value=0,
+        native_max_value=100,
+        mode=NumberMode.SLIDER,
+        async_set_value=lambda router, value: router.set_brightness(brightness=value),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the switch."""
```

Comment:
Number, not switch

Reply:
Oops, revealed my copy paste from the other integrations. My bad lol.